packages/core/src/components/essentials/legend.ts

```diff
@@ -215,9 +215,20 @@ export class Legend extends Component {
 
 				lastYPosition = yPosition;
 
+				// Test if legendItems are placed in the correct direction
+				const testHorizontal = (!legendOrientation ||
```

Comment:
why do we need an extra check? is there any way to make this work without the extra code?

it seems to be working fine right now

Reply:
Because the legend has been called twice. `hasDeactivatedItems` is initially true with `selectedGroups` and the check marks aligned vertically in the first render.  And second render won't run since it's `g.check` is not empty.